### PR TITLE
fix(alert modal): set current user as default alert owner during new alert initialization

### PR DESCRIPTION
**Pull request by anamitraadhikari**
### SUMMARY
Currently, when the modal window opens, the "owners" field is left unpopulated. This is slightly annoying as it would be more user friendly and efficient to automatically pre-populate the field with the currently logged-in user.

To address this, I propose retrieving the current user from Redux state and setting it as the default owner when the alert modal is opened.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
Screenshot 2023-05-15 at 1 43 35 AM

After:
Screenshot 2023-05-15 at 1 43 57 AM

### TESTING INSTRUCTIONS
Open Superset in browser > Click on the "Settings" menu on top right > Click on "Alerts & Reports" > on the "Alerts & Reports" page, click on the "+ ALERT" button to create a new alert & notice the behavior.

### ADDITIONAL INFORMATION
NA

### CODE REVIEW

File: superset-frontend/src/features/alerts/AlertReportModal.tsx [hunk 1]
```diff
@@ -1011,7 +1015,15 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
...
     ) {
-      setCurrentAlert({ ...defaultAlert });
+      setCurrentAlert({
+        ...defaultAlert,
+        owners: [
+          {
+            value: currentUser.userId,
+            label: `${currentUser.firstName} ${currentUser.lastName}`,
+          },
+        ],
```
Comment (villebro) [inline superset-frontend/src/features/alerts/AlertReportModal.tsx:1025]: Note: I was going to propose breaking this out into a util, but it turns out we have a lot of different `User` types that are very different (sometimes `id` us used, other times it's `userId`, and sometimes it's `firstName`, other times `first_name` etc). So best leave this as-is and do a separate PR to clean up all user types in one go.

**Review by villebro — APPROVED** [review]
LGTM, thanks for the improvement!

File: superset-frontend/src/features/alerts/AlertReportModal.tsx (hunk 1 above)
Comment (curious86) [reply superset-frontend/src/features/alerts/AlertReportModal.tsx:1025]: Ack.

**Comment by rusackas** [comment]
Thanks for the PR, and looking forward to the next!
